internal/moreos/moreos_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package moreos
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/getenvoy/internal/test/morerequire"
+)
+
+func TestIsExecutable(t *testing.T) {
+	tempDir, removeTempDir := morerequire.RequireNewTempDir(t)
+	defer removeTempDir()
+
+	bin := filepath.Join(tempDir, "envoy"+Exe)
+	require.NoError(t, os.WriteFile(bin, []byte{}, 0700))
+
+	f, err := os.Stat(bin)
+	require.NoError(t, err)
+
+	require.True(t, isExecutable(f))
+}
+
+func TestIsExecutable_Not(t *testing.T) {
+	tempDir, removeTempDir := morerequire.RequireNewTempDir(t)
+	defer removeTempDir()
+
+	bin := filepath.Join(tempDir, "foo.txt")
+	require.NoError(t, os.WriteFile(bin, []byte{}, 0600))
+
+	f, err := os.Stat(bin)
+	require.NoError(t, err)
+
+	require.False(t, isExecutable(f))
+}
+
+func TestLineSeparator(t *testing.T) {
+	stdout := new(bytes.Buffer)
+	cmd := exec.Command("echo", "cats")
```

Comment:
in windows, we need to have an if statement and do `"cmd.exe", "/c", "echo cats"`. This proves that it works and helps ensure we catch newline things without running envoy